### PR TITLE
Brook.groups

### PR DESCRIPTION
**Pull request by BrookW-83**
Screenshot 2023-06-28 041036
Screenshot 2023-06-28 041113
Screenshot 2023-06-28 041137
Screenshot 2023-06-28 041209

### CODE REVIEW

**Comment by BrookW-83** [comment]
Try to add responsiveness to side bar

File: src/components/groups/CreateGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+    if (!divisionId) {
+      setErrors((prevErrors) => ({
+        ...prevErrors,
+        divisionId: "division level is required",
+      }));
+      return;
+    }
+
+    try {
```
Comment (jebessaD) [inline src/components/groups/CreateGroup.tsx]: Use the default RTK query error to handle if there is a request error

File: src/components/groups/CreateGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+      await createData({
+        name,
+        capacity: capacity as number,
+        telegramLink,
+        divisionId: divisionId as number,
+      }).unwrap();
+      setGroup(initialState);
+      setErrors(initialError);
+    } catch (errors) {}
```
Comment (jebessaD) [inline src/components/groups/CreateGroup.tsx]: If you find that a try-catch block is important and you need to do something after catching an error, perform the necessary actions in the catch block

File: src/components/groups/CreateGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  capacity: "",
+  telegramLink: "",
+  divisionId: "",
+};
+
+const CreateGroup: React.FC<CreateGroup> = ({ handleClose }) => {
+  const [group, setGroup] = useState(initialState);
+  const [createData, { isLoading, error }] = useCreateGroupMutation();
+  const [errors, setErrors] = useState(initialError);
+  const [onCreate, setCreat] = useState<boolean>(false);
```
Comment (jebessaD) [inline src/components/groups/CreateGroup.tsx]: Spelling error `setCreate` 
and make it  
const [showCreate, setShowCreate]

File: src/components/groups/CreateGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        capacity: capacity as number,
+        telegramLink,
+        divisionId: divisionId as number,
+      }).unwrap();
+      setGroup(initialState);
+      setErrors(initialError);
+    } catch (errors) {}
+  };
+
+  const handleCreat = () => {
```
Comment (jebessaD) [inline src/components/groups/CreateGroup.tsx]: Remove unused function

File: src/components/groups/SideBar.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  setSelectedGroup: (value: string) => void;
+}
+
+const Sidebar: React.FC<SidebarProps> = ({ id, setSelectedGroup }) => {
+  const { data, isLoading, error } = useGetGroupByIdQuery(id);
+  const [autoFillGroup, { isLoading: isFilling }] = useAutoFillGroupMutation();
+  const [showPopup, setShowPopup] = useState<Boolean>(false);
+  const [searchValue, setSearchValue] = useState("");
+
+  const handleAutofill = async (id: any) => {
```
Comment (jebessaD) [inline src/components/groups/SideBar.tsx:26]: If you don't need to perform any actions after autofill, you can remove the 'async/await' keywords.

**Review by jebessaD — CHANGES_REQUESTED** [review]
Please ensure that any unused code is removed from the codebase.

**Pull request closed without merging**